pkg/platform/kube/controller/nucliofunction.go

```diff
@@ -551,3 +555,34 @@ func (fo *functionOperator) enrichNodeSelector(ctx context.Context, function *nu
 		"NodeSelector", function.Status.EnrichedNodeSelector)
 	return nil
 }
+
+func (fo *functionOperator) enrichAndValidateServiceAccount(ctx context.Context, function *nuclioio.NuclioFunction) error {
+	projectName, ok := function.Labels[common.NuclioResourceLabelKeyProjectName]
+	if !ok {
+		return errors.New("Function does not have a project label, cannot validate service account")
```

Comment:
Under what condition is this situation expected (if at all)? If there are use-cases where you can run functions not in a project scope then this shouldn't be a critical error.